### PR TITLE
Migrate to Poetry

### PR DESCRIPTION
**Pull request by vanshg**
Migrate to using Poetry (https://python-poetry.org/docs)

- Adds support for Python 3.11, released last week
- Poetry's `pyproject.toml` file supersedes `setup.py`, `setup.cfg`, `MANIFEST.in`, `requirements.txt`, `dev-requirements.txt`
- Updated README with new instructions
- Update GitHub Actions to use poetry

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -17,17 +17,26 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (mldangelo) [inline .github/workflows/test.yml]: this requires an update to the changelog and a major version bump in `smile_id_core/__init__.py`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -17,17 +17,26 @@ jobs:
...
+      uses: snok/install-poetry@v1
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+    - name: Load cached venv
+      id: cached-poetry-dependencies
+      uses: actions/cache@v3
+      with:
+        path: .venv
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
```
Comment (mldangelo) [inline .github/workflows/test.yml]: just pipx install poetry. See: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages


```yml
- name: Install poetry
  run: pipx install poetry
- uses: actions/setup-python@v4
  with:
    python-version: ${{ matrix.python-version }}
    cache: 'poetry'
- run: poetry install
- run: poetry run pytest
```

File: README.md [hunk 3]
```diff
@@ -559,10 +559,10 @@ This will return the smile services endpoint as a json object and  can then be u
 
 ## Development
 
-Reference: https://virtualenv.pypa.io/en/latest/installation.html
-
-1. Set up virtual env.
-2. After checking out the repo, run `pip install -r requirements` to install all required packages.
+1. Ensure you have `poetry` installed: https://python-poetry.org/docs#installation
```
Comment (mldangelo) [inline README.md:562]: this is great, thanks

File: README.md [hunk 4]
```diff
@@ -571,23 +571,27 @@ This is the https://packaging.python.org/tutorials/packaging-projects/ that you
...
+- Bump the version number in `pyproject.toml` and `smile_id_core/__init__.py` 
+- Update `changelog.md` with the new version number and the changes made
+- Run `poetry build` to build package.
+- Ensure your PyPI credentials are propery configured
+- Run `poetry publish` to publish package.
 
 ## Contributing
 
-Bug reports and pull requests are welcome on GitHub at https://github.com/smileidentity/smile-identity-core-python
+Bug reports and pull requests are welcome on GitHub at https://github.com/smileidentity/smile-identity-core-python-3
```
Comment (mldangelo) [inline README.md:591]: thank you

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+[tool.poetry]
+name = "smile-id-core"
+version = "1.0.8"
```
Comment (mldangelo) [inline pyproject.toml]: you should grab this from __init__.py. This is a good thread https://github.com/python-poetry/poetry/issues/144

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+exclude = ["tests/**/*"]
+
+
+[tool.poetry.dependencies]
+python = "^3.7"
+requests = ">=2.24.0"
+pycryptodome = "3.9.8"
+
+[tool.poetry.dev-dependencies]
+pytest = "*"
```
Comment (mldangelo) [inline pyproject.toml]: specify a minimum version here

**Review by mldangelo — COMMENTED** [review]
this is great work. I left a few nits.

File: .github/workflows/test.yml (hunk 2 above)
Comment (vanshg) [reply .github/workflows/test.yml]: Nice, this is much simpler than what I had

File: smile_id_core/__init__.py [hunk 7]
```diff
@@ -1,8 +1,14 @@
...
+try:
+    import importlib.metadata as importlib_metadata
+except ModuleNotFoundError:
+    import importlib_metadata
+
 from smile_id_core.IdApi import IdApi
-from smile_id_core.WebApi import WebApi
-from smile_id_core.Signature import Signature
 from smile_id_core.ServerError import ServerError
+from smile_id_core.Signature import Signature
```
Comment (mldangelo) [inline smile_id_core/__init__.py:9]: thank you for sorting these

**Review by mldangelo — COMMENTED** [review]
This looks great. I'll LGTM as soon as you update the changelog. Thanks a lot @vanshg

File: pyproject.toml (hunk 5 above)
Comment (vanshg) [reply pyproject.toml]: Done

File: pyproject.toml (hunk 6 above)
Comment (vanshg) [reply pyproject.toml]: Done

**Comment by vanshg** [comment]
@mldangelo I pinned `black` to 22.8, so no longer need to drop 3.6 support. Bumped the SDK version to 1.0.9

File: .github/workflows/test.yml (hunk 1 above)
Comment (vanshg) [reply .github/workflows/test.yml]: Restored 3.6 support, so did a minor bump instead due to the added 3.11 support

File: smile_id_core/__init__.py (hunk 7 above)
Comment (vanshg) [reply smile_id_core/__init__.py:9]: IDE auto optimizes imports 🚀

**Review by mldangelo — APPROVED** [review]
Thank you!

**Pull request merged**